wasm/src/language_service.rs

```diff
@@ -33,17 +33,25 @@ impl LanguageService {
         LanguageService(inner)
     }
 
-    pub fn update_document(&mut self, uri: &str, version: u32, text: &str, is_exe: bool) {
-        self.0.update_document(
-            uri,
-            version,
-            text,
-            if is_exe {
-                qsc::PackageType::Exe
-            } else {
-                qsc::PackageType::Lib
-            },
-        );
+    pub fn update_configuration(&mut self, config: IWorkspaceConfiguration) {
+        let config: WorkspaceConfiguration = config.into();
+        self.0
+            .update_configuration(&qsls::protocol::WorkspaceConfigurationUpdate {
+                target_profile: config.targetProfile.map(|s| match s.as_str() {
+                    "base" => qsc::TargetProfile::Base,
```

Comment:
Being in the struct both TargetProfile and PackageType are optional, does this handle the case where those fields might be None/undefined?

Reply:
Yep! The underlying language service is always initialized with the defaults (see the very top of this PR.). If `update_configuration` is never called, or any of the fields are null, those values are simply left alone.